webapp/packages/core-root/src/ServerNodeService.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2023 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { Bootstrap, injectable } from '@cloudbeaver/core-di';
+import { Executor, IExecutor } from '@cloudbeaver/core-executor';
+import { GraphQLService, WsSocketConnectedEvent } from '@cloudbeaver/core-sdk';
+
+import { ServerNodeError } from './ServerNodeError';
+import { ServerEventId, SessionEventSource } from './SessionEventSource';
+
+@injectable()
+export class ServerNodeService extends Bootstrap {
+  applicationRunId = '';
+
+  onApplicationRunIdChange: IExecutor;
+  constructor(
+    private readonly graphQLService: GraphQLService,
+    private readonly sessionEventSource: SessionEventSource
+  ) {
+    super();
+    this.onApplicationRunIdChange = new Executor();
+    this.sessionEventSource.onEvent<WsSocketConnectedEvent>(ServerEventId.CbSessionWebsocketConnected, data => {
+      this.applicationRunIdChanged(data.applicationRunId);
+    });
+  }
+
+  register(): void { }
+
+  load(): void { }
+
+  applicationRunIdChanged(applicationRunId: string): void {
+    if (this.applicationRunId === '') {
+      this.applicationRunId = applicationRunId;
+      return;
+    }
+
+    if (this.applicationRunId === applicationRunId) {
+      return;
+    }
+
+    this.applicationRunId = applicationRunId;
+    this.graphQLService.blockRequests(new ServerNodeError('Server node changed'));
```

Comment:
we need to call new created method in `SessionEventSource` let's call it `disconnect`
`disconnect` method must set private property `disconnected` to `true`

Use this property in the `retryTimer`
```
this.retryTimer = interval(RETRY_INTERVAL)
      .pipe(
        filter(() => !this.sessionExpireService.expired && networkStateService.state && !this.disconnected)
      );
```

Reply:
please also check that front-end will not be trying to open websocket

For example on topic subscriptions:
Open / close "Log Viewer"